resources/todomvc/big-dom-generator/src/components/tree-area.jsx

```diff
@@ -1,56 +1,68 @@
 const TreeItem = (props) => {
-    const { nodeCount, random, maxDepth, maxNumChildren, childProbability, currentDepth } = props;
-    nodeCount.current = nodeCount.current + 4;
-    // Choose whether to have children.
-    const children = random.coin(childProbability) ? <FolderWrapper nodeCount={nodeCount} random={random} maxNumChildren={maxNumChildren} maxDepth={maxDepth} childProbability={childProbability} currentDepth={currentDepth + 1} /> : null;
-    const treeViewItemIsOpen = children && currentDepth < MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH ? "is-open" : "";
+    const { treeNode, currentDepth } = props;
+
+    const isExpandableItem = treeNode.type === "expandableItem";
+    const children = isExpandableItem ? treeNode.children.map((child) => <TreeItem treeNode={child} currentDepth={currentDepth + 1} />) : null;
+    const treeViewItemIsOpen = isExpandableItem && currentDepth < MAX_VISIBLE_TREE_VIEW_ITEM_DEPTH ? "is-open" : "";
     return (
         <li className={`spectrum-TreeView-item ${treeViewItemIsOpen}`}>
             <a className="spectrum-TreeView-itemLink">
-                {children
-                    ? <ChevronRight className="spectrum-Icon spectrum-TreeView-itemIndicator spectrum-TreeView-itemIcon" />
-                    : <TaskListIcon className="task-list-icon spectrum-Icon spectrum-TreeView-itemIndicator spectrum-TreeView-itemIcon spectrum-Icon--sizeM" />
+                {treeNode.type === "nonExpandableItem"
+                    ? <TaskListIcon className="task-list-icon spectrum-Icon spectrum-TreeView-itemIndicator spectrum-TreeView-itemIcon spectrum-Icon--sizeM" />
+                    : <ChevronRight className="spectrum-Icon spectrum-TreeView-itemIndicator spectrum-TreeView-itemIcon" />
                 }
 
-                <span className="just-span spectrum-TreeView-itemLabel">{children ? "Sprint" : "Todo List"}</span>
+                <span className="just-span spectrum-TreeView-itemLabel">{isExpandableItem ? "Sprint" : "Todo List"}</span>
             </a>
-            {children}
+            {isExpandableItem && <ul className="spectrum-TreeView spectrum-TreeView--sizeS">{children}</ul>}
         </li>
     );
 };
 
+/**
+ * TreeItem generates either 6 or 10 elements because ChevronRight is
+ * inlined as a svg element with one path child and TaskListIcon is
+ * inlined as a svg element with 4 path children and one defs child.
+ *
+ * To reach 6 elements, the following dom structure is generated (expandableItem, chevronRight):
+ * <li class="spectrum-TreeView-item is-open">
+ *  <a class="spectrum-TreeView-itemLink">
+ *      <svg>
+ *          <path></path>
+ *      </svg>
+ *      <span>Sprint</span>
+ *  </a>
+ *  <ul></ul>
+ * </li>
+ * To reach 10 elements, the following dom structure is generated (nonExpandableItem, TaskListIcon):
+ * <li class="spectrum-TreeView-item">
+ *   <a class="spectrum-TreeView-itemLink">
+ *     <svg>
+ *       <defs><style>.Smock_TaskList_18_N_svg__fill{fill:#464646}</style></defs>
```

Comment:
nit: can you please put the style element on its own line, this is easier to count :-) (and I believe we don't need the content )

Reply:
Done.